.github/workflows/build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.5', '3.6', '3.7', '3.8']
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install tox-gh-actions
+      - name: Lint with flake8
+        run: |
+          make lint
+      - name: Test with tox
+        run: |
+          make test-all
```

Comment:
I haven't added codecov since I wasn't sure how this is used. Do you have a private repo and hence you need a token for this?

There's an [Action](https://github.com/codecov/codecov-action) that can be used for this. The code would look like below. This would also require creating a GitHub Secret for storing the token.
```
https://github.com/codecov/codecov-action#usage
- name: Upload Coverage
  uses: codecov/codecov-action@v1
  with:
    token: ${{ secrets.CODECOV_TOKEN }}
    file: ./coverage.xml
```